File: api/src/main/java/net/blueshell/api/event/FileEventListener.java
package net.blueshell.api.event;

import net.blueshell.api.common.event.PostRemoveEvent;
import net.blueshell.api.model.File;
import net.blueshell.api.service.FileService;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.transaction.event.TransactionPhase;
import org.springframework.transaction.event.TransactionalEventListener;

@Component
public class FileEventListener {

    private final FileService files;

    public FileEventListener(FileService files) {
        this.files = files;
    }

//    @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
//    @Transactional(propagation = Propagation.REQUIRES_NEW)
//    public void onDelete(PostRemoveEvent<File> evt) {
//        var f = evt.getSource();
//        files.deleteFromStorage(f);
//    }
}

File: services/api/src/main/kotlin/net/blueshell/api/file/domain/FileEventListener.kt
package net.blueshell.api.file.domain

import net.blueshell.api.file.api.FileService
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

@Component
class FileEventListener(
    private val files: FileService
) {
    @EventListener
    @Transactional(propagation = Propagation.REQUIRES_NEW)
    fun onDelete(evt: FileDeleted) {
        files.deleteFromStoragePath(evt.path)
    }
}
